sui-execution/latest/sui-move-natives/src/crypto/nitro_attestation.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::gas_algebra::InternalGas;
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    natives::function::NativeResult,
+    pop_arg,
+    values::{Value, VectorRef},
+};
+use smallvec::smallvec;
+use std::collections::VecDeque;
+use sui_types::nitro_attestation::{parse_nitro_attestation, verify_nitro_attestation};
+
+use crate::{object_runtime::ObjectRuntime, NativesCostTable};
+use move_vm_runtime::native_charge_gas_early_exit;
+
+pub const NOT_SUPPORTED_ERROR: u64 = 0;
+pub const PARSE_ERROR: u64 = 1;
+pub const VERIFY_ERROR: u64 = 2;
+// Gas related structs and functions.
+
+#[derive(Clone)]
+pub struct NitroAttestationCostParams {
+    pub parse_base_cost: Option<InternalGas>,
+    pub parse_cost_per_byte: Option<InternalGas>,
+    pub verify_base_cost: Option<InternalGas>,
+    pub verify_cost_per_cert: Option<InternalGas>,
+}
+
+macro_rules! native_charge_gas_early_exit_option {
+    ($native_context:ident, $cost:expr) => {{
+        use move_binary_format::errors::PartialVMError;
+        use move_core_types::vm_status::StatusCode;
+        native_charge_gas_early_exit!(
+            $native_context,
+            $cost.ok_or_else(|| {
+                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                    .with_message("Gas cost for group ops is missing".to_string())
```

Comment:
Nit: update message to be for nitro attestations here (or feel free to export the other macro and use that here too -- either way is fine with me).